### PR TITLE
Updated Error message for downloadonly in prepare_updates function

### LINKED ISSUE #192: Yum in RHEL <=6.6 doesn't have --downloadonly
When running Auter on a RHEL 6.6 device yum errored out:
WARNING: downloadonly option is not available

When I ran a yum install yum-plugin-downloadonly it upgraded yum instead as a later version of yum includes the downloadonly option.

My advice would be to require a specific version of yum, however I don't know exactly which version included the downloadonly option.  The best I can give you is the version of yum that didn't work:
yum-3.2.29-60.el6.noarch

### PR DESCRIPTION
**Pull request by reaperzn**
Resolves #192

### CODE REVIEW

File: auter.yumdnfModule [hunk 1]
```diff
@@ -87,7 +87,12 @@ function prepare_updates() {
         logit "INFO: No updates are available to be downloaded."
       fi
     else
-      logit "WARNING: downloadonly option is not available"
+      if [[ "${ONLYINSTALLFROMPREP}" == "yes" ]]; then
```
Comment (nrhodes91) [inline auter.yumdnfModule]: This is already nested within the outer `if [[ "${PREDOWNLOADUPDATES}" == "yes" ]]; ...` on  line 48 so will always evaluate to true. Might as well remove the `else`?

The same goes for `auter.aptModule`.

**Comment by rhodesn** [comment]
I don't understand the use case for `PREDOWNLOADUPDATES=yes` and `ONLYINSTALLFROMPREP=no` as having `PREDOWNLOADUPDATES=no` will achieve the same thing if the package manager supports `downloadonly`. If the package manager does not support `downloadonly` and `PREDOWNLOADUPDATES=yes`, there is only one error that needs logging regardless of `ONLYINSTALLFROMPREP`; in this case there is a fundamental conflict between the configuration and what the package manager supports.

```sh
logit "ERROR: DOWNLOADOPTION set to 'yes' but the '--downloadonly' option is not available in the current version of ${PACKAGE_MANAGER}"
```

**Comment by reaperzn** [comment]
The purpose of `PREDOWNLOADUPDATES=yes` and `ONLYINSTALLFROMPREP=no` is to reduce the patching window during the apply stage. This may help to reduce projected maintenance downtime windows. In this case, a warning has been logged in lines https://github.com/rackerlabs/auter/pull/200/files#diff-ae441e278de8c4ba41550f8b5ad008dfR65 and https://github.com/rackerlabs/auter/pull/200/files#diff-b645dff3ceda040dbd3afdb97750c5f9R94 stating that the downloadonly option is not available. This is a warning because at the apply phase, any packages that have updates but not in the cache directory will still be downloaded from the repos due to the `ONLYINSTALLFROMPREP=no` option.

The purpose of this PR is to quit auter when you are in a situation where the package manager does not have the `--downloadonly` option available however auter has been configured to only install the pre-downloaded packages (ie: `PREDOWNLOADUPDATES=yes` and `ONLYINSTALLFROMPREP=yes`). The here is that auter will only attempt to install packages from the download directory (`/var/cache/auter/${profile}/`) which will always be empty if the  `--downloadonly` is not available. This is a hard error because there may actually be valid updates available however auter will currently incorrectly report that there are no updates available when the apply phase is run and exit cleanly.

**Comment by rhodesn** [comment]
Thanks @reaperzn. I know you've explained that before but I have something to come back to in the future now.

**Review by rhodesn — APPROVED** [review]
Testing looks good.

**Pull request merged**